source/slang/slang-emit-cpp.cpp

```diff
@@ -1652,8 +1658,10 @@ void CPPSourceEmitter::_emitWitnessTableDefinitions()
         m_writer->emit(" = {\n");
         m_writer->indent();
         auto seqIdDecoration = witnessTable->findDecoration<IRSequentialIDDecoration>();
-        SLANG_ASSERT(seqIdDecoration);
-        m_writer->emit((UInt)seqIdDecoration->getSequentialID());
+        if (seqIdDecoration)
+            m_writer->emit((UInt)seqIdDecoration->getSequentialID());
+        else
+            m_writer->emit("0");
```

Comment:
Does this `0` case arise in practice? Should it be something where we give an error message?

Reply:
It arises when I am using the code to generate function-pointer based dispatch, where this field is not used and sequential IDs are not assigned.